File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/personal/PersonalDeriveAccountTest.java
package opensdk.sdk.apis.personal;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

import java.io.IOException;
import opensdk.sdk.apis.constant.UrlConstants;
import org.web3j.protocol.klaytn.core.method.response.PersonalDeriveAccountResponse;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.web3j.protocol.http.HttpService;
import org.web3j.protocol.klaytn.Web3j;

@DisplayName("Personal RPC Test")
public class PersonalDeriveAccountTest {
  private Web3j w3 = Web3j.build(new HttpService(UrlConstants.SERVER_URL));
  @Disabled
  @Test
  @DisplayName("RPC personal_deriveAccount")
  void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
    String url = "url";
    String path = "path";
    boolean pin =true;
    PersonalDeriveAccountResponse response = w3.personalDeriveAccount(url, path, pin).send();
    assertNotNull(response);
    assertNull(response.getError());
  }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/personal/PersonalDeriveAccount.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");

const sdk = new OpenSdk.PersonalApi(new OpenSdk.ApiClient(RPC));

describe('personal_deriveAccount API', () => {
    test.skip('should return personal_deriveAccount', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            done();
        };

        const url = "url";
        const path = "path";
        const pin = true;

        sdk.deriveAccount(url, path, {pin}, callbackOne);
    });
});

